### PR TITLE
cgroup refactoring

### PR DESCRIPTION
**Pull request by kolyshkin**
This is a set of patches aiming to prepare the codebase for #2386 

Please review commits one by one.

### CODE REVIEW

**Comment by mrunalp** [comment]
Looks fine but needs rebase.

**Comment by kolyshkin** [comment]
Rebased

**Comment by mrunalp** [comment]
LGTM

**Comment by kolyshkin** [comment]
rebased again :)

File: libcontainer/cgroups/fs/apply_raw.go [hunk 1]
```diff
@@ -61,11 +61,19 @@ type subsystem interface {
...
 	mu       sync.Mutex
-	Cgroups  *configs.Cgroup
-	Rootless bool // ignore permission-related errors
-	Paths    map[string]string
+	cgroups  *configs.Cgroup
+	rootless bool // ignore permission-related errors
+	paths    map[string]string
+}
+
+func NewManager(cg *configs.Cgroup, paths map[string]string, rootless bool) *manager {
```
Comment (AkihiroSuda) [inline libcontainer/cgroups/fs/apply_raw.go:71]: The return type should be a public interface

File: libcontainer/cgroups/systemd/v1.go [hunk 2]
```diff
@@ -17,10 +17,17 @@ import (
...
-type LegacyManager struct {
+type legacyManager struct {
 	mu      sync.Mutex
-	Cgroups *configs.Cgroup
-	Paths   map[string]string
+	cgroups *configs.Cgroup
+	paths   map[string]string
+}
+
+func NewLegacyManager(cg *configs.Cgroup, paths map[string]string) *legacyManager {
```
Comment (AkihiroSuda) [inline libcontainer/cgroups/systemd/v1.go:26]: ditto: should return a public interface

**Comment by AkihiroSuda** [comment]
Looks good but a couple of nits

File: libcontainer/cgroups/systemd/v1.go (hunk 2 above)
Comment (kolyshkin) [reply libcontainer/cgroups/systemd/v1.go:26]: Yeah, I have noticed the discrepancy. Apparently it's OK to return a private struct as long as it conforms to the interface. Anyway, I'll unify to return cgroups.Manager everywhere

Comment (kolyshkin) [reply libcontainer/cgroups/systemd/v1.go:26]: fixed

Comment (AkihiroSuda) [reply libcontainer/cgroups/systemd/v1.go:26]: `NewLegacyManager` doesn't seem fixed

Comment (kolyshkin) [reply libcontainer/cgroups/systemd/v1.go]: fixed

File: libcontainer/cgroups/fs/apply_raw.go (hunk 1 above)
Comment (kolyshkin) [reply libcontainer/cgroups/fs/apply_raw.go]: fixed

**Comment by AkihiroSuda** [comment]
LGTM

**Comment by mrunalp** [comment]
LGTM

**Pull request merged**